Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -481,6 +507,37 @@ class HistoryPanel: SiteTableViewController, HomePanel {
     }
 }
 
+extension HistoryPanel: UITableViewDataSourcePrefetching {
+    func tableView(_ tableView: UITableView, prefetchRowsAt indexPaths: [IndexPath]) {
+        guard !isFetchInProgress, indexPaths.contains(where: isLoadingCell) else {
+            return
+        }
+
+        fetchData().uponQueue(.main) { result in
+            if let sites = result.successValue {
+                let indexPaths: [IndexPath] = sites.compactMap({ site in
+                    guard let site = site, let latestVisit = site.latestVisit else {
+                        return nil
+                    }
+
+                    let indexPath = self.groupedSites.add(site, timestamp: TimeInterval.fromMicrosecondTimestamp(latestVisit.date))
+                    return IndexPath(row: indexPath.row, section: indexPath.section + 1)
+                })
+
+                self.tableView.insertRows(at: indexPaths, with: .automatic)
+            }
+        }
+    }
+
+    func isLoadingCell(for indexPath: IndexPath) -> Bool {
```

Comment:
nit: `shouldLoadCell` or `shouldLoad(indexPath: IndexPath)`, isLoadingCell confused me as I thought this was a check to see if something was in a 'loading' state.